### ISSUE TITLE
Deleting a Feature crashes Rekit

### ISSUE THREAD

**Issue by bradley-varol** [supnate/rekit#88]
I have tested this twice. Deleting a Feature causes the following error:

`ls: no such file or directory: <PATH>/src/features/<FeatureName>/*.js
ls: no such file or directory: <PATH>/src/features/<FeatureName>/redux/*.js
Error: Can't find such file: <PATH>/src/features/<FeatureName>/route.js`

**Comment by supnate** [comment]
What's your command? I can't reproduce the error.

**Comment by bradley-varol** [comment]
I created a new rekit project with `rekit create --clean`, ran `npm install`, `npm start`, created a new Feature and then deleted it from inside the GUI. That is all.

**Comment by supnate** [comment]
What's your feature name? I also can't reproduce with GUI. And Windows or MacOS?

**Comment by bradley-varol** [comment]
I'm trying to test for you once more but I actually can't run `npm start` - [screenshot](http://prntscr.com/igv5hy).

**Comment by supnate** [comment]
That's because port 6077 is in use. Please close the running instance and retry.

**Comment by supnate** [comment]
I don't have Windows running aside, will check it later.

**Comment by bradley-varol** [comment]
Oops lol, sorry. Okay, I tested again for you. I opened the GUI, created a feature called "Test Feature", deleted the feature and got the same error.

`ls: no such file or directory: F:/abc123/ABC123/src/features/test-feature/*.js`
`ls: no such file or directory: F:/abc123/ABC123/src/features/test-feature/redux/*.js`
`Error: Can't find such file: F:/abc123/ABC123/src/features/test-feature/route.js`

**Comment by supnate** [comment]
Okay, thanks for this. I will find a Windows and check this issue.

**Comment by zeroxx1986** [comment]
I think this issue is connected to this one:
[Under Windows, renaming a feature breaks the app rebuild while app is running](https://github.com/supnate/rekit/issues/83)

Reason is that the files which are to be deleted / renamed are locked by the system / Windows. If you rename/delete a feature via CLI commands and You're not running the development server, everything is working.

**Comment by bradley-varol** [comment]
I think it most probably is connected too. Is this something that can be fixed?

**Comment by supnate** [comment]
Thanks @zeroxx1986 for the information. It's strange that permission issue happens on Windows system. Maybe need fix from your side by allowing the user who runs node to manage files of the project.

**Comment by zeroxx1986** [comment]
I think the problem resides in webpack, or somewhere similar. The locking of files happens due to webpack watching the directory recursively for changes happening to files which are connected to the project. In my opinion this watch is actually causing the files to be locked by the system, preventing them to be deleted/renamed/moved (basically any operation that alters the file's path). Unfortunately this cannot be fixed with any kind of permission setting...

**Comment by zeroxx1986** [comment]
I did a quick digging around, and found traces of the same issue at Webpack: [Webpack development server locks files when running under Windows](https://github.com/webpack/webpack/issues/2070)
The devs say that the watching functionality is actually coming from the `chokidar` library. I think this is the same error: [Deleting folder results in EPERM error that locks all folders that are recursively watched](https://github.com/paulmillr/chokidar/issues/643)

**Comment by bradley-varol** [comment]
Is there something that I can do to work around this issue in the meantime?

**Comment by zeroxx1986** [comment]
yes, you need to stop the development server when you either delete, move or rename a feature. How to do that if you've stopped the dev server? Via the rekit CLI commands, which can be invoked even when the dev server is not running.

**Comment by bradley-varol** [comment]
Great, thank you.
